pkg/cri/server/sandbox_run.go

```diff
@@ -290,7 +291,22 @@ func (c *criService) RunPodSandbox(ctx context.Context, r *runtime.RunPodSandbox
 
 		// Update network namespace in the container's spec
 		c.updateNetNamespacePath(spec, sandbox.NetNSPath)
-		if err := updateContainerSpec(ctx, container, spec); err != nil {
+
+		if err := container.Update(ctx,
+			// Update spec of the container
+			func(ctx context.Context, client *containerd.Client, c *containers.Container) error {
+				any, err := typeurl.MarshalAny(spec)
+				if err != nil {
+					return fmt.Errorf("failed to marshal spec %+v: %w", spec, err)
+				}
+				c.Spec = any
+				return nil
+			},
```

Comment:
```suggestion
			containerd.UpdateContainerOpts(containerd.WithSpec(spec)),
```

Or if that's not to your preference, you could also simplify the code here using `protobuf.MarshalAnyToProto(s)` instead of the `typeurl.MarshalAny` dance.

Reply:
Good to know. That looks much better.